samples/nrf_desktop/src/hw_interface/buttons.c

```diff
@@ -31,29 +32,61 @@ enum state {
 	STATE_SUSPENDING
 };
 
-
-static const u8_t col_pin[] = { 2, 21, 20, 19};
-static const u8_t row_pin[] = {29, 31, 22, 24};
-
-static struct device *gpio_dev;
-static struct gpio_callback gpio_cb;
+static struct device *gpio_devs[GPIO_PORT_NUM];
+static struct gpio_callback gpio_cb[GPIO_PORT_NUM];
 static struct k_delayed_work matrix_scan;
 static enum state state;
 static struct k_spinlock lock;
 
+#if CONFIG_BOARD_NRF52840_PCA20041
+static const u8_t col_pin[] = {2, 21, 20, 19};
+static const u8_t col_port[] = {0, 0, 0, 0};
+static const u8_t row_pin[] = {29, 31, 22, 24};
+static const u8_t row_port[] = {0, 0, 0, 0};
+#elif CONFIG_BOARD_NRF52840_PCA10059
+static const u8_t col_pin[] = {0};
+static const u8_t col_port[] = {0};
+static const u8_t row_pin[] = {SW0_GPIO_PIN};
+static const u8_t row_port[] = {1};
+#else
+#error "Buttons not supported on this platform"
+#endif
+
+#define IS_MATRIX (ARRAY_SIZE(col_pin) > 1)
+
+static void scan_fn(struct k_work *work);
+
+
+static bool port_used(u8_t port)
```

Comment:
`port_is_used()`

Reply:
this function won't be needed - check comment below